trading/exchanges/websockets_exchanges/websocket/cryptofeed_websocket.py

```diff
@@ -0,0 +1,321 @@
+        self.is_handling_recent_trades = False
+        self.is_handling_funding = False
+
+        self.channels = []
+        self.callbacks = {}
+
+    def init_web_sockets(self, time_frames, trader_pairs):
+        self.cryptofeed_handler = FeedHandler()
+        self.exchange_class = self.__class__._get_cryptofeed_class(self.exchange_manager.exchange_class_string.title())
+        self.trader_pairs = trader_pairs
+        self.time_frames = time_frames
+        self.converted_traded_pairs = self._get_converted_traded_pairs(trader_pairs)
+
+        if self.converted_traded_pairs:
+            self.add_recent_trade_feed()
+            self.add_order_book_feed()
+            self.add_tickers_feed()
+            self.add_ohlcv_feed()
+            # self.add_funding_feed()
+
+            # ensure feeds are added
+            self._create_cryptofeed_feeds()
+        else:
+            self.logger.warning(f"{self.exchange_manager.exchange_class_string.title()}'s "
+                                f"websocket has no symbol to feed")
+
+    # Feeds
+    def add_recent_trade_feed(self):
+        if self._is_feed_available(TRADES):
+            self._add_feed_and_run_if_required(TRADES, TradeCallback(RecentTradesCallBack(self).recent_trades_callback))
+            self.is_handling_recent_trades = True
+        else:
+            self.logger.warning(f"{self.exchange_manager.exchange_class_string.title()}'s "
+                                f"websocket is not handling recent trades")
+
+    def add_order_book_feed(self):
+        if self._is_feed_available(L2_BOOK):
+            self._add_feed_and_run_if_required(L2_BOOK, BookCallback(OrderBookCallBack(self).l2_order_book_callback))
+            self.is_handling_order_book = True
+        else:
+            self.logger.warning(f"{self.exchange_manager.exchange_class_string.title()}'s "
+                                f"websocket is not handling order book")
+
+    def add_tickers_feed(self):
+        if self._is_feed_available(TICKER):
+            self._add_feed_and_run_if_required(TICKER, TickerCallback(TickersCallBack(self).tickers_callback))
+            self.is_handling_price_ticker = True
+        else:
+            self.logger.warning(f"{self.exchange_manager.exchange_class_string.title()}'s "
+                                f"websocket is not handling tickers")
+
+    def add_ohlcv_feed(self):
+        if self._is_feed_available(TRADES):
+            for time_frame in self.time_frames:
+                self._add_feed_and_run_if_required(
+                    TRADES,
+                    OHLCV(Callback(OHLCVCallBack(self, time_frame).ohlcv_callback),
+                          window=self._convert_time_frame_minutes_to_seconds(TimeFramesMinutes[time_frame])))
+
+            # add real time handler
+            self._add_feed_and_run_if_required(
+                TRADES,
+                OHLCV(Callback(OHLCVCallBack(self, 0).ohlcv_callback), window=0))
```

Comment:
is it necessary ? i don't understand why we need a 0 timeframe

Reply:
It's not a 0 timeframe, cryptofeed provide "windows" that corresponds to seconds between updates. So here I specified 0s to handle all updates.